xlators/storage/posix/src/posix-inode-fd-ops.c

```diff
@@ -2529,6 +2570,9 @@ posix_release(xlator_t *this, fd_t *fd)
     VALIDATE_OR_GOTO(this, out);
     VALIDATE_OR_GOTO(fd, out);
 
+    if (fd->inode->fd_count == 1)
```

Comment:
It's unfortunate that `release` function is called before decrementing the `fd_count` of the inode. This makes impossible to do this in a safe way. For example, if there are two concurrent releases, `fd_count` will be 2, so none of the threads will unlink the renamed file.

I think we should modify `fd_destroy()` in `libglusterfs/src/fd.c` so that the `fd_count` is updated before calling `release`, but there's a comment there to only decrease the count after having closed the file. I'm not sure if this means after having called `release`.

@pranithk you added that comment. Is it safe to decrement before calling `release` ? is there any known dependency ?

Reply:
@xhernandez In inode there are 2 variables that keep track of fd-count.
This is my understanding
1) fd_count: Tracks the opens fds on the disk file system
     Is incremented on fd_bind() and decremented after fd_destroy()
2) active_fd_count: Tracks the fd_t structures that are active. This number will be decremented before calling fd_destroy(). I think eager-lock required this subtle difference.

Reply:
@xhernandez I went through the code again just now. It doesn't look like we need these two with subtle differences. We can just keep the behavior of active-fd-count and things should work as expected. For backward compatibility we should send the same count in both GET_OPEN_FD_COUNT and GET_ACTIVE_OPEN_FD_COUNT

Reply:
I have moved fd decrement part before calling release in fd_destroy() and ran shard suit. The suit was successful. I will push the changes and run the regression.

Reply:
Could you also remove active_fd_count variable in inode and just keep fd_count?

Reply:
sure @pranithk

Reply:
@pranithk @xhernandez Changes are done to the fd_unref() and fd_destroy() function.  

Sorry that I forcefully updated the patch.